pkg/controllers/customcluster_controller.go

```diff
@@ -201,20 +195,38 @@ func (r *CustomClusterController) Reconcile(ctx context.Context, req ctrl.Reques
 			return ctrl.Result{}, nil
 		}
 		log.Error(err, "failed to find customMachine", "customMachine", customMachinekey)
-		return ctrl.Result{RequeueAfter: RequeueAfter}, err
+		return ctrl.Result{}, err
+	}
+
+	// Fetch the KubeadmControlPlane instance.
+	kcpKey := client.ObjectKey{
+		Namespace: cluster.Spec.ControlPlaneRef.Namespace,
+		Name:      cluster.Spec.ControlPlaneRef.Name,
+	}
+	kcp := &controlplanev1.KubeadmControlPlane{}
+	if err := r.Client.Get(ctx, kcpKey, kcp); err != nil {
+		if apierrors.IsNotFound(err) {
+			log.Info("kcp does not exist", "kcp", kcpKey)
+			return ctrl.Result{}, nil
+		}
+		log.Error(err, "failed to get kcp", "kcp", kcpKey)
+		return ctrl.Result{}, err
 	}
 
 	patchHelper, err1 := patch.NewHelper(cluster, r.Client)
 	if err1 != nil {
-		return ctrl.Result{RequeueAfter: RequeueAfter}, errors.Wrapf(err1, "failed to init patch helper for cluster %s", req.NamespacedName)
+		return ctrl.Result{}, errors.Wrapf(err1, "failed to init patch helper for cluster %s", req.NamespacedName)
 	}
 
 	defer func() {
 		if err := r.Status().Update(ctx, customCluster); err != nil {
 			log.Error(err, "failed to update customCluster status", "customCluster", req)
 			reterr = err
 		}
-
+		if err := r.Update(ctx, customCluster); err != nil {
```

Comment:
we donot need calling updater status and update

The patch below can work

Reply:
you are right, the patch works